icon docs (#1661)
NuGet/docs.microsoft.com-nuget: docs/reference/errors-and-warnings/NU5046.md

---
title: NuGet Error NU5046
description: NU5046 Error code
author: dominoFire
ms.author: feaguila
ms.date: 9/4/2019
ms.topic: reference
ms.reviewer: karann
f1_keywords: 
  - NU5046
---

# NuGet Error NU5046

<pre>The icon file 'Icon.png' does not exist in the package.</pre>

# Issue

> NuGet is unable find the local file that is marked as the package icon.

Reviewer: Calling it a `local file` makes it sound like it's looking on the filesystem, not in the package, but the error text in the `<pre>` block, and most of the solution makes it sound like the error is when the icon is not in the package.

How about:
```suggestion
NuGet is unable find the icon file in the package.
```

# Solution

> - Make sure that the file that is marked as the package icon exists and it is readable.

Reviewer: Again, this sentence makes it sound like it's looking for the file on the filesystem, not in the package. I also find the term "marked" a bit strange and would prefer if the sentence could be changed to avoid it.

In fact, is this point even needed? If the file doesn't exist on the local file system, but the nuspec has a `file` element, or the csproj has a `<None Include="icon.jpg" />` element, but the file doesn't exist on the filesystem (or isn't readable), wouldn't pack return a different error code? Unreadable file trying to be included in the package isn't icon specific.

> - Make sure that the file that is marked as the package icon exists and it is readable.

Author: ```suggestion
- Make sure that the file that is marked as the package icon exists at the source and it is readable, and the target matches the path expected by the `icon` property.
```

- Ensure that the file is referenced in the nuspec or in the project file.

>   * When you are creating a package from a nuspec file, make sure to include the icon file in the `<files/>` section:

Reviewer: For both examples below, consider removing everything not specific to the icon and put in a placeholder like `...` to signify that extra content is necessary for a full, working example, but is unrelated to this specific problem. I can imagine beginners finding the extra elements a distraction from understanding how to fix their problem.

Reviewer 2: What if users want a complete working example?

Author: we have a separate sample section.

    ```xml

>     <package>

Reviewer: the XML has inconsistent indenting (children of the `metadata` and `files` elements are not indented).

>     <package>

Reviewer: We want to encourange customers to use our pack targets, rather than using a nuspec file. So, the MSBuild example should be first. Same with the other file.

        <metadata>
        <id>sample.lib</id>
        <version>0.0.1</version>
        <authors>nuget</authors>
        <owners>nuget</owners>
        <license type="expression">MIT</license>
        <description>Package containing the icon file</description>
        <icon>icon.jpg</icon>
        </metadata>
        <files>
        <file src="icon.jpg" />
        </files>
    </package>
    ```

  * When creating a package from a MSBuild project file, make sure to reference the icon file in the project, as follows:

    ```xml
    <Project Sdk="Microsoft.NET.Sdk">

>         <PropertyGroup>

Reviewer: Children of `PropertyGroup` and `ItemGroup` are not indented.
